src/passes/MergeBlocks.cpp

```diff
@@ -277,6 +277,22 @@ struct MergeBlocks : public WalkerPass<PostWalker<MergeBlocks>> {
   void visitStore(Store* curr) {
     optimize(curr, curr->value, optimize(curr, curr->ptr), &curr->ptr);
   }
+  void visitAtomicRMW(AtomicRMW* curr) {
+    optimize(curr, curr->value, optimize(curr, curr->ptr), &curr->ptr);
+  }
+  // XXX TODO: why doesn't this work for select?
```

Comment:
I don't see how this could cause an infinite loop, that's odd. But, this code was actually wrong, there is a fuzz fix in #1095 for it. Let's land that first, it's possible doing this on the fixed code will work. If not, we can investigate the infinite loop there (i.e., might not make sense to debug the infinite loop on code that is being replaced anyhow).